python/ray/serve/_private/deployment_state.py

```diff
@@ -1912,6 +1933,39 @@ def _check_curr_status(self) -> Tuple[bool, bool]:
 
         return False, any_replicas_recovering
 
+    def _has_reached_steady_state(self) -> bool:
+        """Whether or not this deployment has reached steady-state."""
+
+        target_version = self._target_state.version
+        num_replicas_running_at_target_version = self._replicas.count(
+            states=[ReplicaState.RUNNING], version=target_version
+        )
+
+        autoscaling_policy = self._target_state.info.autoscaling_policy
+        if autoscaling_policy is None:
+            has_reached_steady_state = (
+                self.curr_status_info.status_trigger
+                != DeploymentStatusTrigger.CONFIG_UPDATE_STARTED
+                or num_replicas_running_at_target_version
+                == self._target_state.target_num_replicas
+            )
```

Comment:
This branch is not used (or tested). I added it for completeness' sake, but it may be better to remove it altogether. I'm not sure what to replace it with though.

Reply:
I replaced the branch with an `assertion`.